src/net_processing.cpp

```diff
@@ -278,9 +278,20 @@ class PeerManagerImpl final : public PeerManager
     void SetBestHeight(int height) override { m_best_height = height; };
     void Misbehaving(const NodeId pnode, const int howmuch, const std::string& message) override;
     void ProcessMessage(CNode& pfrom, const std::string& msg_type, CDataStream& vRecv,
-                        const std::chrono::microseconds time_received, const std::atomic<bool>& interruptMsgProc) override;
+                        const std::chrono::microseconds time_received, const std::atomic<bool>& interruptMsgProc) override LOCKS_EXCLUDED(m_mutex_message_handling);
 
 private:
+    /** Message handling mutex.
+     *  Message processing is single-threaded, so anything only accessed
+     *  by ProcessMessage(s) or SendMessages can be guarded by this mutex,
+     *  which guarantees it's only accessed by a single thread.
```

Comment:
How guarding with a mutex guarantees "it's only accessed by a single thread"? Is the intention to say "it's only accessed by a single thread _at a given point in time_"? If yes, then this applies to any mutex and the comment better be removed to avoid confusion. If it means that the mutex guarantees "it's only accessed by _the_ single thread _ThreadMessageHandler()_", then it is not true because any other thread can grab the mutex and access the relevant variables.